### ISSUE TITLE
potential fix leads to compilation error (IDE0059)

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#54037]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/potential-fix-leads-to-compilation-error/1445654)._

---
If you are not using car instance the following line

`Car car = new(something);`

turns to

`_ = new(something);`

after quick fix.

But should be

`_ = new Car(something);`

---
#### Fiona Niu[MSFT] on 6/9/2021, 08:20 PM: 

<p>Thank you for your feedback! Does this reproduce for all project or specific project? If it’s reproduce with the specific project, could you please provide a sample solution that has this problem so that we can conduct further research?</p>
<p>We look forward to hearing from you!</p>

#### Clemens Schneider on 6/10/2021, 05:34 AM: 

<pre><code><span class="hljs-keyword">namespace</span> <span class="hljs-title">ConsoleApp</span>
{
    <span class="hljs-keyword">class</span> <span class="hljs-title">Program</span>
    {
        <span class="hljs-function"><span class="hljs-keyword">static</span> <span class="hljs-keyword">void</span> <span class="hljs-title">Main</span>(<span class="hljs-params"><span class="hljs-keyword">string</span>[] args</span>)</span>
        {
            Car car = <span class="hljs-keyword">new</span>(); <span class="hljs-comment">// Do quick fix here...</span>
        }
    }

    <span class="hljs-keyword">class</span> <span class="hljs-title">Car</span>
    {

    }
}
</code></pre>

#### Clemens Schneider on 6/10/2021, 05:39 AM: 

<p>namespace ConsoleApp<br>
{<br>
class Program<br>
{<br>
static void Main(string[] args)<br>
{<br>
Car car = new(); // Do quick fix here…<br>
}<br>
}</p>
<pre><code>class Car
{

}
</code></pre>
<p>}</p>

#### Feedback Bot on 6/10/2021, 10:44 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by CyrusNajmabadi** [comment]
Yeah, i believe this is the fix for when a variable isn't used and is turned into a discard.  We need to eitehr make it not run on inferred types, or we need it to insert a cast on the right (or a type name for the object creation expression).

Other examples where you can rnu into this are:  `Func<int, int> identity = a => a;`  This cannot be changed into `_ = a => a;`

**Comment by CyrusNajmabadi** [comment]
I don't get this.  It properly generates the following for me:

![image](https://user-images.githubusercontent.com/4564579/198369948-90e8357b-deee-49ee-b873-21eeef0d9cca.png)

Closing out.
